nilearn/decomposition/tests/test_single_pca.py

```diff
@@ -0,0 +1,26 @@
+import numpy as np
+import nibabel
+
+from nilearn.decomposition.single_pca import SinglePCA
+
+
+def test_single_pca():
+    shape = (6, 8, 10, 5)
+    affine = np.eye(4)
+    rng = np.random.RandomState(0)
+
+    # Create a "multi-subject" dataset
+    data = []
+    for i in range(8):
+        this_data = rng.normal(size=shape)
+        # Create fake activation to get non empty mask
+        this_data[2:4, 2:4, 2:4, :] += 10
+        data.append(nibabel.Nifti1Image(this_data, affine))
+
+    mask_img = nibabel.Nifti1Image(np.ones(shape[:3], dtype=np.int8), affine)
+    single_pca = SinglePCA(mask=mask_img, n_components=3)
+
+    components = single_pca.fit(data).components_
+    assert(isinstance(components, list))
+    assert(components[0].shape == (3, 6 * 8 * 10))
```

Comment:
We could check that the components are reasonably orthogonal: the off-diagonal element of np.dot(components, components.T) being small.

That would make a numerical test, on top of the smoke tests above.